images/builder/docker/custom-docker-builder/build.sh

```diff
@@ -1,4 +1,20 @@
 #!/bin/bash
+
+# See https://docs.openshift.org/latest/creating_images/custom.html#custom-builder-image
+# for the list of environment variables set by OpenShift before the custom
+# builder image is run.
+#
+# Although set as part of the API, the environment variables
+# SOURCE_REPOSITORY/SOURCE_URI, SOURCE_CONTEXT_DIR and SOURCE_REF can also be
+# derived from the BUILD environment variable using `jq`.  If necessary, this
+# technique can be used for extracting other values from the BUILD json from a
+# shell script.
+#
+# SOURCE_REPOSITORY=$(jq -nr '(env.BUILD|fromjson).spec.source.git.uri')
+# SOURCE_URI=$(jq -nr '(env.BUILD|fromjson).spec.source.git.uri')
```

Comment:
`SOURCE_REPOSITORY` and `SOURCE_URI` will have the same values. Looks like, `SOURCE_REPOSITORY` should extract `spec.source.git.repository` or another member.

Reply:
They are documented to be identical (API guarantee means they couldn't be changed).  https://docs.openshift.org/latest/creating_images/custom.html#custom-builder-image.